File: 2657-find-the-prefix-common-array-of-two-arrays/2657-find-the-prefix-common-array-of-two-arrays.go
func findThePrefixCommonArray(A []int, B []int) []int {
    n := len(A)
    ans := make([]int, n)
    vis := make([]int, n+1)
    commonCount := 0
    
    for i := 0; i < n; i++ {
        // Increment visit count for A[i]
        vis[A[i]]++
        if vis[A[i]] == 2 {
            commonCount++
        }
        
        // Increment visit count for B[i]
        vis[B[i]]++
        if vis[B[i]] == 2 {
            commonCount++
        }
        
        ans[i] = commonCount
    }
    
    return ans
}

File: 2657-find-the-prefix-common-array-of-two-arrays/2657-find-the-prefix-common-array-of-two-arrays.java
class Solution {
    public int[] findThePrefixCommonArray(int[] A, int[] B) {
        int n = A.length;
        int[] C = new int[n];
        int[] count = new int[n + 1];
        int commonCount = 0;
        
        for (int i = 0; i < n; i++) {
            count[A[i]]++;
            if (count[A[i]] == 2) {
                commonCount++;
            }
            
            count[B[i]]++;
            if (count[B[i]] == 2) {
                commonCount++;
            }
            
            C[i] = commonCount;
        }
        
        return C;
    }
}